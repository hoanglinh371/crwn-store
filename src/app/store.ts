import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';

import cartReducer from '../features/cart/redux/cart.slice';
import categoryReducer from '../features/category/redux/category.slice';
import productReducer from '../features/product/redux/product.slice';

export const store = configureStore({
  reducer: {
    productReducer,
    cartReducer,
    categoryReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
