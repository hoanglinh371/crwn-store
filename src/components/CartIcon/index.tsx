import React from 'react';
import classNames from 'classnames/bind';
import { ReactComponent as ShoppingIcon } from 'src/assets/shopping-bag.svg';

import styles from './CartIcon.module.scss';

const cx = classNames.bind(styles);

const CartIcon: React.FC = () => {
    return (
        <div className={cx('container')}>
            <ShoppingIcon className={cx('shopping-icon')} />
            <span className={cx('item-count')}>{0}</span>
        </div>
    );
};

export default CartIcon;
