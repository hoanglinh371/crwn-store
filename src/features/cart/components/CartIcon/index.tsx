import React from 'react';
import classNames from 'classnames/bind';
import { ReactComponent as ShoppingIcon } from 'src/assets/shopping-bag.svg';
import { useAppSelector, useAppDispatch } from 'src/app/hooks';

import { selectCartCount } from '../../redux/cart.selectors';
import { toggleCartDropdown } from '../../redux/cart.slice';
import styles from './CartIcon.module.scss';

const cx = classNames.bind(styles);

const CartIcon: React.FC = () => {
  const dispatch = useAppDispatch();
  const cartCount = useAppSelector(selectCartCount);

  const handleToggleCartDropdown = () => {
    dispatch(toggleCartDropdown());
  };

  return (
    <div className={cx('container')} onClick={handleToggleCartDropdown}>
      <ShoppingIcon className={cx('shopping-icon')} />
      <span className={cx('item-count')}>{cartCount}</span>
    </div>
  );
};

export default CartIcon;
