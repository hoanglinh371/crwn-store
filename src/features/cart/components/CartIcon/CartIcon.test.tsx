import { cleanup, render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from 'src/app/store';

import CartIcon from '.';

describe('CartIcon component', () => {
  afterEach(cleanup);

  it('Should_MatchSnapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <CartIcon />
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });
});
