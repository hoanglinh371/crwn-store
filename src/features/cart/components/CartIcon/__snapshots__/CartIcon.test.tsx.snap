// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartIcon component Should_MatchSnapshot 1`] = `
<div>
  <div
    class="container"
  >
    <svg
      class="shopping-icon"
    >
      shopping-bag.svg
    </svg>
    <span
      class="item-count"
    >
      0
    </span>
  </div>
</div>
`;
